autoagora/model_builder.py

```diff
@@ -27,6 +32,63 @@ async def apply_default_model(subgraph: str):
     await set_cost_model(subgraph, model)
 
 
+# (mrq) stands for multi root query
+async def mrq_model_builder(subgraph: str, pgpool: asyncpg.Pool) -> str:
+    logs_db = LogsDB(pgpool)
+    # Obtain most queried mrq
+    most_frequent_multi_root_queries = (
+        await logs_db.get_most_frequent_queries_null_time(subgraph)
+    )
+    for mrq_info in most_frequent_multi_root_queries:
+        await obtain_query_time(subgraph, mrq_info, logs_db)
+    # Call tables with info created
+    most_frequent_queries = await logs_db.get_most_frequent_queries(
+        subgraph, mrq_table=True
+    )
+    model = build_template(subgraph, most_frequent_queries)
+    return model
+
+
+# Obtains the execution time for n amount of random queries
+async def obtain_query_time(
+    subgraph, multi_root_query_info, logs_db, iterations: int = 100
+):
+    # Call db to obtain related variable lists
+    query_log_ids = await logs_db.get_query_logs_id(
+        multi_root_query_info.hash
+    )
+    no_queries = len(query_log_ids)
+    for _ in range(iterations):
+
+        query_id = query_log_ids[random.randint(0, no_queries-1)][0]
+        query_variables = await logs_db.get_query_variables(query_id)
+        query_vars = {f"_{i}": var for i, var in enumerate(query_variables)}
+
+        async with Client(
+            transport=AIOHTTPTransport(args.graph_node_query_endpoint),
+            fetch_schema_from_transport=False,
+        ) as session:
+
+            start_q_execution = time.time()
+            await session.execute(gql(multi_root_query_info.query), variable_values=query_vars)  # type: ignore
+            multi_root_query_info.query_time_ms = int(
+                (time.time() - start_q_execution) * 1000
+            )
+
+        multi_root_query_info.timestamp = datetime.now()
+        logs_db.save_generated_aa_query_values(
+            multi_root_query_info, subgraph, query_variables
+        )
+
+
+async def mrq_model_update_loop(subgraph: str, pgpool):
+    while True:
+        model = await mrq_model_builder(subgraph, pgpool)
+        await set_cost_model(subgraph, model)
+        # Should this also be this amount of time?
+        await aio.sleep(args.relative_query_costs_refresh_interval)
```

Comment:
This should be a different time, otherwise it would be too slow.
Also consider using `random.lognormvariate`: https://www.desmos.com/calculator/k3en6o5ikf

Reply:
What would be a good "default" value and apply lognormvariate to it, to give it some "randomness"?

Reply:
The values in the link should be a good start